pkg/agg/damage/damage.go

```diff
@@ -105,7 +105,12 @@ func (b *buffer) Add(result stats.Result) {
 			charTargetDPS[ev.Target] += ev.Damage
 			charElementDPS[ev.Element] += ev.Damage
 			charDPS += ev.Damage
-			sourceDPS[ev.Source] += ev.Damage
+			sourceDPSKey := ev.Source
+			reactModifier := string(ev.ReactionModifier)
+			if reactModifier != "" {
+				sourceDPSKey += " (" + reactModifier + ")"
```

Comment:
I'm not a fan of this approach. If in the future we ever add proper game mappings for keys, we will be unable to easily leverage them in UI for localization since we would need to string parse to get the original sourceDPSKey. Ideally, data model should be changed to add react modifier as a new dimension and then UI can decide if it wants to merge like so or add this as a toggle on the graph or what. The main problem with this is if you change the data structure for this data, you make the new UI incompatible with the older version on source dps data unless you also program backwards compatibility which is a pain in the ass and imo not worth it unless you have a clear idea on how you'd leverage this extra information without concatenating in UI like you're doing here.

At minimum: you should make the react modifier more human readable and have some hand crafted map of what the user-facing keys here should be

Reply:
did the minimum without adding a handcrafted map since I think the string representation of ReactionModifier is good enough (vaporize/melt/spread/aggravate)